torch/csrc/jit/codegen/cuda/lower_insert_syncs.cpp

```diff
@@ -0,0 +1,58 @@
+#include <torch/csrc/jit/codegen/cuda/ir_iostream.h>
+#include <torch/csrc/jit/codegen/cuda/lower_utils.h>
+
+#include <torch/csrc/jit/codegen/cuda/lower_insert_syncs.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+void SyncInserter::handle(kir::ForLoop* fl) {
+  bool prev_needs_sync = needs_sync;
+  active_scope = fl;
+
+  for (auto expr : fl->body().exprs()) {
+    handle(expr);
+  }
+
+  if (needs_sync && !fl->iter_domain()->isThread() &&
+      fl->body().exprs().back()->getExprType().value() != ExprType::Sync) {
+    fl->body().push_back(new kir::Sync());
+  }
+
+  bool needs_sync = prev_needs_sync;
```

Comment:
Is this supposed to declare a new variable? Or is it supposed to assign `prev_needs_sync` to the existing `needs_sync`?